p2p/background/router.go

```diff
@@ -342,13 +350,30 @@ func (rtr *backgroundRouter) bootstrap(ctx context.Context) error {
 			return nil
 		}
 
-		if err := rtr.host.Connect(ctx, libp2pAddrInfo); err != nil {
+		if err := rtr.connectWithRetry(ctx, libp2pAddrInfo); err != nil {
```

Comment:
Can you elaborate on the rationale behind this change?

Reply:
1. #892 is a 500 line PR that I haven't had a chance to dig into yet and I'm not 100% sure we'll be able to get it in by end-of-week

2. I briefly looked at https://github.com/pokt-network/pocket/commit/1f05989cf6022a6d1fdb709a00430bea961859f7 but need a bit more time to think/digest.

3. I believe the two changes are still complementary and `connectWithRetry` is a fast (and logical) way to go around the race condition, which personally unblocked my development as well.

Lmk wyt